src/sources/docker_logs.rs

```diff
@@ -1352,233 +1359,261 @@ mod integration_tests {
 
-        let log = events[0].as_log();
-        assert_eq!(log[log_schema().message_key()], message.into());
-        assert_eq!(log[super::CONTAINER], id.into());
-        assert!(log.get(super::CREATED_AT).is_some());
-        assert_eq!(log[super::IMAGE], "busybox".into());
-        assert!(log.get(format!("label.{}", label).as_str()).is_some());
-        assert_eq!(events[0].as_log()[&super::NAME], name.into());
-        assert_eq!(
-            events[0].as_log()[log_schema().source_type_key()],
-            "docker".into()
-        );
+            let log = events[0].as_log();
+            assert_eq!(log[log_schema().message_key()], message.into());
+            assert_eq!(log[super::CONTAINER], id.into());
+            assert!(log.get(super::CREATED_AT).is_some());
+            assert_eq!(log[super::IMAGE], "busybox".into());
+            assert!(log.get(format!("label.{}", label).as_str()).is_some());
+            assert_eq!(events[0].as_log()[&super::NAME], name.into());
+            assert_eq!(
+                events[0].as_log()[log_schema().source_type_key()],
+                "docker".into()
+            );
+        })
+        .await;
     }
 
     #[tokio::test]
     async fn include_image() {
         trace_init();
 
-        let message = "15";
-        let name = "vector_test_include_image";
-        let config = DockerLogsConfig {
-            include_containers: Some(vec![name.to_owned()]),
-            include_images: Some(vec!["busybox".to_owned()]),
-            ..DockerLogsConfig::default()
-        };
+        assert_source_compliance(&SOURCE_TAGS, async {
+            let message = "15";
+            let name = "vector_test_include_image";
+            let config = DockerLogsConfig {
+                include_containers: Some(vec![name.to_owned()]),
+                include_images: Some(vec!["busybox".to_owned()]),
+                ..DockerLogsConfig::default()
+            };
 
-        let out = source_with_config(config);
+            let out = source_with_config(config).await;
 
-        let docker = docker(None, None).unwrap();
+            let docker = docker(None, None).unwrap();
 
-        let id = container_log_n(1, name, None, message, &docker).await;
-        let events = collect_n(out, 1).await;
-        container_remove(&id, &docker).await;
+            let id = container_log_n(1, name, None, message, &docker).await;
+            let events = collect_n(out, 1).await;
+            container_remove(&id, &docker).await;
 
-        assert_eq!(
-            events[0].as_log()[log_schema().message_key()],
-            message.into()
-        );
+            assert_eq!(
+                events[0].as_log()[log_schema().message_key()],
+                message.into()
+            );
+        })
+        .await;
     }
 
     #[tokio::test]
     async fn not_include_image() {
         trace_init();
 
+        // No assert_source_compliance here since we aren't including the image
+        // no events are collected.
```

Comment:
@neuronull would this be a place to use the assert error helper?

Reply:
It could be! "No events are collected" makes me unsure if the source is actually run and an error condition occurs?

But, its real easy to try.... Actually, I was going to suggest using `assert_source_error()` in the same way `assert_source_compliance()` was used above, but I just realized that for sources I only created `run_and_assert_source_error()`  hmmm.
I'm not sure that format of function will fit the needs of this test case.

@StephenWakely  if you want to have a go at implementing `assert_source_error()` I  think it should be pretty straight forward, it would probably look like this:

```Rust
  /// Convenience wrapper for running source error tests.   
  #[track_caller]  
  pub async fn assert_source_error<T>(tags: &[&str], f: impl Future<Output = T>) -> T {    
      assert_source(&COMPONENT_TESTS_ERROR, tags, f).await     
  } 
```        

But if it ends up not being straightforward, I can tackle it in a separate PR.

Reply:
There is nothing really running. Since we are excluding the container there are no events collected, so it doesn't error and it doesn't collect events... It just does nothing.

Reply:
Ah, I misunderstood the purpose the test. I'm good to go on this then 😄